o11y/o11y.go

```diff
@@ -117,6 +133,8 @@ type Span interface {
 	// End sets the duration of the span and tells the related provider that the span is complete,
 	// so it can do its appropriate processing. The span should not be used after End is called.
 	End()
+
+	//Link(Span)
```

Comment:
is this intended to be excluded from the interface? I see there's a note that it specifically won't be implemented for libhoney directly

Reply:
ah - another refactor cleanup.

This is the problem of getting distracted on code that you started months ago :(